daffodil-core/src/main/scala/org/apache/daffodil/dsom/RestrictionUnion.scala

```diff
@@ -73,11 +74,17 @@ final class Restriction(xmlArg: Node, val simpleTypeDef: SimpleTypeDefBase)
 
   lazy val baseQName = {
     val baseQNameNodeSeq = xml \ "@base"
-    val baseQNameString = baseQNameNodeSeq.text
-    val tryBaseQName = QName.resolveRef(baseQNameString, xml.scope,
+    val tryBaseQName = QName.resolveRef(baseQNameNodeSeq.text, xml.scope,
       tunable.unqualifiedPathStepPolicy)
-    Assert.invariant(tryBaseQName.isSuccess)
-    tryBaseQName.get
+    val result = tryBaseQName.isFailure && baseQNameNodeSeq.text.contains(" ") match {
+      case true => {
+        println("Warning: Invalid spaces in base name removed '" + baseQNameNodeSeq.text + "'")
```

Comment:
This makes sense, I'll rework it in this direction.

Reply:
Actually, it looks like we already have the logic for checking if a qname is valid--QName.resolveRef returns a [``Try``](https://www.scala-lang.org/api/current/scala/util/Try.html) that contains failure information. So the right fix here is rather than doing ``Assert.invariant(tryBaseQName.isSuccess)`` we should throw an SDE if the the ``Try`` failed. There are various ways to throw an SDE, but in this case you can probably do something like:
```scala
schemaDefinitionUnless(tryBaseQName.isSuccess, "some error message about invalid qname base")
```

Reply:
Sounds good. So don't try to remove the spaces, just throw the SDE.